internal/configs/action.go

```diff
@@ -422,3 +367,198 @@ func (a *Action) ProviderConfigAddr() addrs.LocalProviderConfig {
 		Alias:     a.ProviderConfigRef.Alias,
 	}
 }
+
+// decodeActionTriggerRef decodes and does basic validation of the Actions
+// expression list inside a resource's ActionTrigger block, ensuring each only
+// reference a single action. This function was largely copied from
+// decodeReplaceTriggeredBy, but is much more permissive in what References are
+// allowed.
+func decodeActionTriggerRef(expr hcl.Expression) ([]ActionRef, hcl.Diagnostics) {
```

Comment:
For `replace_triggered_by` we validate the allowed index variables during decoding, because we can statically check for `each.key` and `count.index`. A problem with the `import` implementation is that it delays the validation so if the suer tried something unexpected they get a more vague (and kind of incorrect) `single static variable reference is required`.  It doesn't block functionality of this PR, but might be something to improve on later.